src/helpers/proofs.js

```diff
@@ -0,0 +1,237 @@
+  for (let i = 0; i < block.transactions.length; i++) {
+    const siblingTx = block.transactions[i]
+    const path = rlp.encode(siblingTx.transactionIndex)
+    const rawSignedSiblingTx = getTxBytes(siblingTx)
+    await new Promise((resolve, reject) => {
+      txTrie.put(path, rawSignedSiblingTx, err => {
+        if (err) {
+          reject(err)
+        } else {
+          resolve()
+        }
+      })
+    })
+  }
+
+  // promise
+  return new Promise((resolve, reject) => {
+    txTrie.findPath(
+      rlp.encode(tx.transactionIndex),
+      (err, rawTxNode, reminder, stack) => {
+        if (err) {
+          return reject(err)
+        }
+
+        if (reminder.length > 0) {
+          return reject(new Error("Node does not contain the key"))
+        }
+
+        const prf = {
+          blockHash: utils.toBuffer(tx.blockHash),
+          parentNodes: stack.map(s => s.raw),
+          root: getRawHeader(block).transactionsTrie,
+          path: rlp.encode(tx.transactionIndex),
+          value: rlp.decode(rawTxNode.value)
+        }
+        resolve(prf)
+      }
+    )
+  })
+}
+
+export function verifyTxProof(proof) {
+  const path = proof.path.toString("hex")
+  const value = proof.value
+  const parentNodes = proof.parentNodes
+  const txRoot = proof.root
+  try {
+    var currentNode
+    var len = parentNodes.length
+    var nodeKey = txRoot
+    var pathPtr = 0
+    for (var i = 0; i < len; i++) {
+      currentNode = parentNodes[i]
+      const encodedNode = Buffer.from(
+        utils.sha3(rlp.encode(currentNode)),
+        "hex"
+      )
+      if (!nodeKey.equals(encodedNode)) {
+        return false
+      }
+      if (pathPtr > path.length) {
+        return false
+      }
+      switch (currentNode.length) {
+        case 17: // branch node
+          if (pathPtr === path.length) {
+            if (currentNode[16] === rlp.encode(value)) {
+              return true
+            } else {
+              return false
+            }
+          }
+          nodeKey = currentNode[parseInt(path[pathPtr], 16)] // must === sha3(rlp.encode(currentNode[path[pathptr]]))
+          pathPtr += 1
+          break
+        case 2:
+          pathPtr += nibblesToTraverse(
+            currentNode[0].toString("hex"),
+            path,
+            pathPtr
+          )
+          if (pathPtr === path.length) {
+            // leaf node
+            if (currentNode[1].equals(rlp.encode(value))) {
+              return true
+            } else {
+              return false
+            }
+          } else {
+            // extension node
+            nodeKey = currentNode[1]
+          }
+          break
+        default:
+          // all nodes must be length 17 or 2
+          return false
+      }
+    }
+  } catch (e) {
+    return false
+  }
+  return false
+}
+
+export function getReceiptBytes(receipt) {
+  let status = receipt.root
+  if (receipt.status !== undefined && receipt.status !== null) {
+    status = receipt.status
```

Comment:
` status = receipt.status ? "0x01" : "0x00" ` -- will work or why do we need to check for `else if` condition ?

Reply:
@shahbhavir  would love your reviews here.

Reply:
if `receipt.status` has been assigned a value, then the code pushed is correct, or else your suggestion would work too